### ISSUE TITLE
mockBase raising 'size limit exceeded' exception when search size_limit = 0 (default)

### ISSUE THREAD

**Issue by djlambert** [cannatag/ldap3#518]
Using the dev branch. On a connection with ```raise_exceptions=True``` when a search is perfomed on a mock directory and no ```size_limit``` value is passed to ```Connection.search()``` an exception is raised.

```
self = <tests.models.directory.entry.ldap.active_directory.active_directory_entry_test.ActiveDirectoryEntryTest object at 0x12fc2cb38>, session = <sqlalchemy.orm.session.Session object at 0x12f84c5c0>, ad_directory = <tests.mocks.ldap_mocker.LdapMocker object at 0x13066d828>

    def test_add_entries(self, session: orm.Session, ad_directory: ActiveDirectory):
        ad_directory.connect(username='CN=Derek Lambert,OU=Users,DC=domain,DC=local', password='good password')
        session.add(ad_directory)
    
>       for entry in ad_directory.get_entries():

tests/models/directory/entry/ldap/active_directory/active_directory_entry_test.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
audit_tools/models/directory/ldap/ldap.py:154: in get_entries
    for connection in self._paged_search(**search_context):
audit_tools/models/directory/ldap/ldap.py:71: in _paged_search
    **kwargs)
venv/lib/python3.6/site-packages/ldap3/core/connection.py:776: in search
    response = self.post_send_search(self.send('searchRequest', request, controls))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ldap3.strategy.mockSync.MockSyncStrategy object at 0x12fc62860>
payload = (2, 'searchRequest', <SearchRequest value object at 0x130a3ec50 tagSet=<TagSet object at 0x10b1083c8 tags 64:32:3> sub...object at 0x131863e80 tagSet <TagSet object at 0x10b094e48 tags 0:0:4> encoding utf-8 payload [0x3006020203e80400]>]>])

    def post_send_search(self, payload):
        message_id, message_type, request, controls = payload
        self.connection.response = []
        self.connection.result = dict()
        if message_type == 'searchRequest':
            responses, result = self.mock_search(request, controls)
            for entry in responses:
                response = search_result_entry_response_to_dict(entry, self.connection.server.schema, self.connection.server.custom_formatter, self.connection.check_names)
                response['type'] = 'searchResEntry'
                ###
                if self.connection.empty_attributes:
                    for attribute_type in request['attributes']:
                        attribute_name = str(attribute_type)
                        if attribute_name not in response['raw_attributes'] and attribute_name not in (ALL_ATTRIBUTES, ALL_OPERATIONAL_ATTRIBUTES, NO_ATTRIBUTES):
                            response['raw_attributes'][attribute_name] = list()
                            response['attributes'][attribute_name] = list()
                            if log_enabled(PROTOCOL):
                                log(PROTOCOL, 'attribute set to empty list for missing attribute <%s> in <%s>',
                                    attribute_type, self)
                    if not self.connection.auto_range:
                        attrs_to_remove = []
                        # removes original empty attribute in case a range tag is returned
                        for attribute_type in response['attributes']:
                            attribute_name = str(attribute_type)
                            if ';range' in attribute_name.lower():
                                orig_attr, _, _ = attribute_name.partition(';')
                                attrs_to_remove.append(orig_attr)
                        for attribute_type in attrs_to_remove:
                            if log_enabled(PROTOCOL):
                                log(PROTOCOL,
                                    'attribute type <%s> removed in response because of same attribute returned as range by the server in <%s>',
                                    attribute_type, self)
                            del response['raw_attributes'][attribute_type]
                            del response['attributes'][attribute_type]
                ###
                self.connection.response.append(response)
            result = search_result_done_response_to_dict(result)
            result['type'] = 'searchResDone'
            self.connection.result = result
            if self.connection.raise_exceptions and result and result['result'] not in DO_NOT_RAISE_EXCEPTIONS:
                if log_enabled(PROTOCOL):
                    log(PROTOCOL, 'operation result <%s> for <%s>', result, self.connection)
>               raise LDAPOperationResult(result=result['result'], description=result['description'], dn=result['dn'], message=result['message'], response_type=result['type'])
E               ldap3.core.exceptions.LDAPSizeLimitExceededResult: LDAPSizeLimitExceededResult - 4 - sizeLimitExceeded - None - size limit exceeded - searchResDone - None

venv/lib/python3.6/site-packages/ldap3/strategy/mockSync.py:93: LDAPSizeLimitExceededResult
```

Coming from this comparison:
https://github.com/cannatag/ldap3/blob/14e7925abe769a655cadf6cc45333785bd8beb94/ldap3/strategy/mockBase.py#L673

**Comment by cannatag** [comment]
Hi, can you check the code in dev? It should work now. 

Thanks,
Giovanni

**Comment by djlambert** [comment]
It's working now in dev. Thanks!

**Comment by cannatag** [comment]
Fixed in 2.5.
